sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -923,7 +923,13 @@ private[client] class Shim_v2_0 extends Shim with Logging {
       tableName: String,
       throwException: Boolean): Table = {
     recordHiveCall()
-    val table = hive.getTable(dbName, tableName, throwException)
+    val table = try {
+      hive.getTable(dbName, tableName, throwException)
+    } catch {
+      // Hive may have bugs and still throw an exception even if `throwException` is false.
+      case e: HiveException if !throwException && e.getMessage.contains("Unable to fetch table") =>
```

Comment:
Is it ok to skip 2nd condition as throwException=false means the caller side requires suppressing expressions？